tests/FakeItEasy.Analyzer.VisualBasic.Tests/NonVirtualSetupAnalyzerTests.cs

```diff
@@ -0,0 +1,378 @@
+        {
+            const string Test = @"Imports System
+Namespace FakeItEasy.Analyzer.VisualBasic.Tests
+    Class TheClass
+        Sub Test()
+            Dim foo = A.Fake(Of Foo)()
+            A.CallTo(Function() foo.Bar()).Returns(42)
+        End Sub
+    End Class
+
+    Public Class Foo
+        Public Overridable Function Bar() As Integer
+            Throw New NotImplementedException()
+        End Function
+    End Class
+End Namespace
+";
+
+            this.VerifyVisualBasicDiagnostic(Test);
+        }
+
+        [Fact]
+        public void Diagnostic_Should_Not_Be_Triggered_For_Unfortunately_Named_Call()
+        {
+            // Ensure only FakeItEasy triggers the diagnostic
+            const string Test = @"Imports System
+Namespace AnalyzerPrototypeSubjectConfusion
+    Class UnfortunatelyNamedClass
+        Sub TheTest()
+            Dim foo = New Foo()
+            A.CallTo(Sub() foo.Bar(String.Empty))
+        End Sub
+
+        Class A
+            Friend Shared Sub CallTo(ByVal fake As Action)
+                Throw New NotImplementedException()
+            End Sub
+        End Class
+    End Class
+
+    Public Class Foo
+        Public Sub Bar(ByVal name As String)
+            Throw New NotImplementedException()
+        End Sub
+    End Class
+End Namespace
+";
+
+            this.VerifyVisualBasicDiagnostic(Test);
+        }
+
+        [Fact]
+        public void Diagnostic_Should_Not_Be_Triggered_For_Nested_Non_Virtual_Member()
+        {
+            const string Test = @"Imports FakeItEasy
+Namespace TheNamespace
+    Class TheClass
+        Sub Test()
+            Dim aClass = New AClass()
+            Dim foo = A.Fake(Of IFoo)()
+            A.CallTo(Function() foo.Bar(aClass.Method())).Returns(42)
+        End Sub
+    End Class
+
+    Class AClass
+        Public Function Method() As Integer
+            Return 3
+        End Function
+    End Class
+    Interface IFoo
+        Function Bar(ByVal i As Integer) As Integer
+    End Interface
+End Namespace";
+
+            this.VerifyVisualBasicDiagnostic(Test);
+        }
+
+        [Fact]
+        public void Diagnostic_Should_Be_Triggered_For_Non_Virtual_Member()
+        {
+            const string Test = @"Imports System
+Namespace FakeItEasy.Analyzer.VisualBasic.Tests
+    Class TheClass
+        Sub Test()
+            Dim foo = A.Fake(Of Foo)()
+            A.CallTo(Function() foo.Bar()).Returns(42)
+        End Sub
+    End Class
+
+    Public Class Foo
+        Public Function Bar() As Integer
+            Throw New NotImplementedException()
+        End Function
+    End Class
+End Namespace
+";
+
+            this.VerifyVisualBasicDiagnostic(
+                Test,
+                new DiagnosticResult
+                {
+                    Id = DiagnosticDefinitions.NonVirtualSetupSpecification.Id,
+                    Message =
+                        "Non virtual member 'Bar' cannot be intercepted.",

```

Comment:
This should really be a comment on the resource that specifies the string, but a user writing in VB _may_ not be familiar with "Non virtual". I have not investigated the difficulty of doing this, but perhaps they'd prefer to hear about a "Non overridable"  member?

(Aside: I've never really looked at VB.NET before, and I'm perfectly comfortable with "abstract" and "virtual" and whatnot, but if I were a programming tyro, I think the VB syntax would be much more helpful. "Overridable" already means something to an English speaker, just like "Shared" comes closer to expressing the concept than "static". Ditto "MustOverride"/"abstract". We've been artificially raising the bar to entry! Because Java, I expect.)